tests/test_overrides_parser.py

```diff
@@ -61,9 +66,12 @@ def eq(item1: Any, item2: Any) -> bool:
         pytest.param("null", None, id="value:null"),
         # int
         pytest.param("1", 1, id="value:int:pos"),
+        pytest.param("+1", 1, id="value:int:explicit_pos"),
+        pytest.param("1___0___", "1___0___", id="value:int:not_an_int"),
```

Comment:
No, you should quote instead (`"'1_000'"`), though Hydra's is handling quotes in a special way I don't really understand (instead of `"1_000"` this yields `QuotedString(text="1_000", quote=Quote.single)`).

Reply:
The quotes have to be re-constructed in some cases so the parsed result contains information about them.